cmake/org.eclipse.cdt.cmake.core/src/org/eclipse/cdt/cmake/core/internal/CMakeToolChainManager.java

```diff
@@ -156,20 +156,18 @@ public void addToolChainFile(ICMakeToolChainFile file) {
 	public void removeToolChainFile(ICMakeToolChainFile file) {
 		init();
 		fireEvent(new CMakeToolChainEvent(CMakeToolChainEvent.REMOVED, file));
-		String tcId = makeToolChainId(file.getProperty(CMakeBuildConfiguration.TOOLCHAIN_TYPE),
-				file.getProperty(CMakeBuildConfiguration.TOOLCHAIN_ID));
-		filesByToolChain.remove(tcId);
-
-		String n = ((CMakeToolChainFile) file).n;
-		if (n != null) {
-			Preferences prefs = getPreferences();
-			Preferences tcNode = prefs.node(n);
-			try {
+		try {
+			String tcId = makeToolChainId(file.getToolChain());
```

Comment:
Yes, this looks like a definite improvement.

I couldn't figure out why the original code was written the way it was.

I think in the past I've worked around this by assigning the TOOLCHAIN_TYPE and TOOLCHAIN_ID properties in my code that adds the toolchain file, just after calling org.eclipse.cdt.cmake.core.ICMakeToolChainManager.newToolChainFile(Path), like this:

			// toolchainFile.setProperty(ICBuildConfiguration.TOOLCHAIN_TYPE, tc.getTypeId());
			// toolchainFile.setProperty(ICBuildConfiguration.TOOLCHAIN_ID, tc.getId());

But this new code improves that situation so it's no longer necessary.

Reply:
Thanks @betamaxbandit for having a look. Marking this thread as resolved.